handlers/dimension/listHandler.go

```diff
@@ -0,0 +1,26 @@
+package dimension
+
+import (
+	"encoding/json"
+	"net/http"
+	"github.com/ONSdigital/dp-dd-api-stub/models"
+	"fmt"
+)
+
+func listHandler(w http.ResponseWriter, req *http.Request) {
+	datasetID := req.URL.Query().Get(":datasetId")
+	dimensionID := req.URL.Query().Get(":dimensionId")
+
+	dimension, err := findDimension(datasetID, dimensionID)
+	if err != nil {
+		fmt.Println(err)
+		response, _ := json.Marshal(models.Dimension{})
+		w.WriteHeader(400)
+		w.Write(response)
+		return
+	}
+
+	jsonEncoder := json.NewEncoder(w)
+	jsonEncoder.Encode(dimension)
+	w.WriteHeader(200)
```

Comment:
Good point. I moved it around

```
	dimension, err := findDimension(datasetID, dimensionID)
	if err != nil {
		fmt.Println(err)
		response, _ := json.Marshal(models.Dimension{})
		w.WriteHeader(400)
		w.Write(response)
		return
	}

	w.WriteHeader(200)
	jsonEncoder := json.NewEncoder(w)
	jsonEncoder.Encode(dimension)
```

`json.Marshal()` has been called from `findDimension()` method so I believe it can safely be encoded.